### DISCUSSION TITLE
Vaadin's awesome a11y tools

### DISCUSSION THREAD

**Discussion by bennypowers** [RedHat-UX/red-hat-design-system#754] [General]
Hey, check out the awesome a11y tools that Vaadin's developing. Our friend @web-padawan drew my attention, and we should ~~totally steal these~~ take inspiration or even import them

https://github.com/vaadin/web-components/tree/main/packages/a11y-base

**Comment by bennypowers** [comment]
check this out @nikkimk @hellogreg

**Reply by web-padawan** [reply]
Hey, thanks for sharing this stuff! Here are a few more links:

- [Slides](https://docs.google.com/presentation/d/1avZO5MFD5h045C9sxBtj9nuswawMVLJnCaiRlX_TIXU/edit?usp=sharing) that I shared publicly on [Mastodon](https://fosstodon.org/@kulykov/109955855103074959) and [Twitter](https://twitter.com/serhiikulykov/status/1631261550290825220)
- [Open Lit meeting notes](https://docs.google.com/document/d/1QtHrZpjLhkMhMkuL3h63VaHOp4E20_3sj1JX1q0zrt4/edit?usp=sharing) where we discussed this topic
- [wc-hooks](https://github.com/devongovett/wc-hooks) by @devongovett from [React Aria](https://github.com/adobe/react-spectrum/tree/main/packages/%40react-aria) @adobe

**Reply by web-padawan** [reply]
Let me also add a brief summary of what I shared, as the meeting wasn't recorded.

## Prior art and inspiration

- We started two years ago with an audit by @TetraLogical @patrickhlauke
- Since then, we moved `<input>` to light DOM inspired by [Lion](https://lion-web.netlify.app/) @dakMor
- Got some inspiration also from [Spectrum Web Components](https://opensource.adobe.com/spectrum-web-components/) @westbrook
- Another case of well-thought a11y is [SAP UI5 Web Components](https://sap.github.io/ui5-webcomponents/playground/advanced/accessibility/) @ilhan007
- Last but not least, @carbon-design-system is [really cool](https://github.com/carbon-design-system/carbon-for-ibm-dotcom) @ariellalgilmore
- Last year we had a [webinar](https://vaadin.com/blog/web-components-design-systems-and-accessibility-recording-available) on design systems + a11y hosted by @rolfsmeds

## Current project scope

- We are currently working on a 6 weeks project (+2 weeks cooldown) for a11y
- This is a third iteration of a11y effort at @vaadin since Vaadin 22.0 and 23.0
- My personal focus at the moment is proper overlay modality for screen readers
- Started by porting [`aria-hidden`](https://github.com/theKashey/aria-hidden) by @theKashey to avoid React dependency
- Other focus areas include restoring focus to the button when dialog closes, etc
- Our great colleagues @rolfsmeds and @jouni are working on High Contrast stuff
- Vaadin Design team members @anezthes and @juuso-vaadin also help us a lot

## Closing words

To all the awesome folks out there: firstly, my apologies to poke you all. Hope you don't mind 😇 
I do understand how overwhelmed we are with all the GitHub notifications from time to time.

The project that @bennypowers shared is something I'm super committed to at the moment.
Quite a few @vaadin customers are strongly interested in a11y (especially in the EU area).

If anyone has ideas on how we could collaborate, please contact me at serhii@vaadin.com 🙏

**Reply by patrickhlauke** [reply]
Nice one 👍 

Also, credit to @IanPouncey on the @TetraLogical side
